octodns/manager.py

```diff
@@ -274,21 +269,14 @@ def _build_kwargs(self, source):
         return kwargs
 
     def configured_sub_zones(self, zone_name):
-        # Reversed pieces of the zone name
-        pieces = zone_name[:-1].split('.')[::-1]
-        # Point where at the root of the tree
+        name = zone_name[:-1]
         where = self.zone_tree
-        # Until we've hit the bottom of this zone
-        try:
-            while pieces:
-                # Point where at the value of our current piece
-                where = where[pieces.pop(0)]
-        except KeyError:
-            self.log.debug('configured_sub_zones: unknown zone, %s, no subs',
-                           zone_name)
-            return set()
-        # We're not pointed at the dict for our name, the keys of which will be
-        # any subzones
+        while True:
+            parent = next(filter(lambda k: name.endswith(k), where), None)
+            if not parent:
+                break
+            where = where[parent]
```

Comment:
Fwiw, using list comprehensions over next & filter is preferred in py3, though both are a little cumbersome here.

If my read/understanding is correct something like this may work:

```suggestion
            parents = [k for k in where if name.endswith(k)]
            if not parents:
                break
            where = where[parent[0]]
```

Reply:
There will only ever be one parent, thus my goal was to have the search be lazy and to stop attempting string comparisons once the parent was found.

If you're okay with keeping the `next()`?
If so, I could use a list comprehension as follows, that will keep that short-circuit, which I think is cleaner as it doesn't require indexed access to the value on line 278 and 279.
```
parent = next((k for k in where if name.endswith(k)), None)
```

Reply:
I've used this in the follow commit adding comments and switching to list comprehensions.

Reply:
That seems good. More inline with what py3 prefers.